graalpython/com.oracle.graal.python/src/com/oracle/graal/python/builtins/objects/foreign/ForeignObjectBuiltins.java

```diff
@@ -1031,6 +1031,28 @@ protected static long doIt(Object object,
                 gil.acquire();
             }
         }
+        @Specialization(limit = "3", guards="lib.fitsInBigInteger(object)")
```

Comment:
Should this have a `!lib.fitsInLong(object)` guard also? Otherwise, this branch may be taken even if the `object` would fit in long, just before previous runs specialized to the biginteger branch.

Because `fitsInLong` apparently seems to imply `fitsInBigInteger`, as shown in example below:
```
jshell> var v = Value.asValue(42)
v ==> 42

jshell> v.fitsInLong()
$51 ==> true

jshell> v.fitsInBigInteger()
$52 ==> true
```

Reply:
I've expanded the test in 8fb2b586a7d7d936e5b086cdc918727f465aaa4c to verify everything works with as expected. I haven't modified the code.